packages/mermaid/src/docs/syntax/classDiagram.md

```diff
@@ -145,7 +145,7 @@ class BankAccount{
 
 Generics can be representated as part of a class definition, and for class members/return types. In order to denote an item as generic, you enclose that type within `~` (**tilde**). **Nested** type declarations such as `List<List<int>>` are supported, though generics that include a comma are currently not supported. (such as `List<List<K, V>>`)
 
-Generics can be represented as part of a class definition and also in the parameters or the return value of a method/function (note that any references to the class later in diagram code is used without the type )
+> _note_ when a generic is used within a class definition, the generic type is NOT considered part of the class name. i.e.: for any syntax which required you to reference the class name, you need to drop the type part of the definition. This also means that mermaid does not currently support having two classes with the same name, but different generic types.
```

Comment:
We have a custom syntax for notes that you can use:

````suggestion
```note
When a generic is used within a class definition, the generic type is NOT considered part of the class name. i.e.: for any syntax which required you to reference the class name, you need to drop the type part of the definition. This also means that mermaid does not currently support having two classes with the same name, but different generic types.
```
````

It will make it look something like the following:

![image](https://github.com/mermaid-js/mermaid/assets/19716675/ef0d3c52-e56a-4e9f-94a1-d6c806d6edbb)


See https://mermaid.js.org/community/documentation.html#how-to-contribute-to-documentation for more info.